Failure to Upgrade to Illegal Version

As end-user I would like to be protected against application of accidentally applied upgrades to illegal versions to avoid having a broken installation left.

Accidentally(!), I started an upgrade of metalctl to version `v0.12.18` (this version does not exist). This failed metalctl and made the current installation failing for further upgrades.

### Steps to reproduce:

```shell
# Upgrade to latest version, as of 4th of July 2024
$ sudo metalctl update do
checksum:b81efa4d3fae1f5e2239797c9c4dbcee
40.27 MiB / 40.27 MiB [---------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 40.68 MiB p/s 1.2s

# Run accidental "down/upgrade" to possibly illegal version string
$ sudo metalctl update do --version v0.12.18
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x15c623c]

goroutine 1 [running]:
github.com/metal-stack/updater.latestRelease({0xc0000570b0, 0x14}, {0x1bb89ba, 0xb}, {0x1bb3e1a, 0x8}, 0xc0002b1c70)
	/home/runner/go/pkg/mod/github.com/metal-stack/updater@v1.2.1/github.go:43 +0x1dc
github.com/metal-stack/updater.New({0x1bb89ba, 0xb}, {0x1bb3e1a, 0x8}, {0x1bb3e1a, 0x8}, 0xc000493c70)
	/home/runner/go/pkg/mod/github.com/metal-stack/updater@v1.2.1/updater.go:35 +0xac
github.com/metal-stack/metalctl/cmd.newUpdateCmd.func2(0xc000021b08?, {0x1bae52d?, 0x4?, 0x1bae4a9?})
	/home/runner/work/metalctl/metalctl/cmd/update.go:50 +0xfb
github.com/spf13/cobra.(*Command).execute(0xc000021b08, {0xc0003b7de0, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002ca908)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/metal-stack/metalctl/cmd.Execute()
	/home/runner/work/metalctl/metalctl/cmd/root.go:52 +0xaa
main.main()
	/home/runner/work/metalctl/metalctl/main.go:8 +0xf

# Trial to run upgrade as non-root.
$ metalctl update do
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x15c623c]

goroutine 1 [running]:
github.com/metal-stack/updater.latestRelease({0xc0005bc138, 0x14}, {0x1bb89ba, 0xb}, {0x1bb3e1a, 0x8}, 0xc0006b81d0)
	/home/runner/go/pkg/mod/github.com/metal-stack/updater@v1.2.1/github.go:43 +0x1dc
github.com/metal-stack/updater.New({0x1bb89ba, 0xb}, {0x1bb3e1a, 0x8}, {0x1bb3e1a, 0x8}, 0xc0006b81d0)
	/home/runner/go/pkg/mod/github.com/metal-stack/updater@v1.2.1/updater.go:35 +0xac
github.com/metal-stack/metalctl/cmd.newUpdateCmd.func2(0xc000301b08?, {0x1bae52d?, 0x4?, 0x1bae4a9?})
	/home/runner/work/metalctl/metalctl/cmd/update.go:50 +0xfb
github.com/spf13/cobra.(*Command).execute(0xc000301b08, {0x2cb3140, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002ce908)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/metal-stack/metalctl/cmd.Execute()
	/home/runner/work/metalctl/metalctl/cmd/root.go:52 +0xaa
main.main()
	/home/runner/work/metalctl/metalctl/main.go:8 +0xf

# Trial to run upgrade as privileged user
$ sudo metalctl update do
Error: Get "http://localhost:8080/cloud/v1/version": dial tcp 127.0.0.1:8080: connect: connection refused
```

### Expected Behavior:
Trial to upgrade/downgrade to illegal version will be replied with appropriate error message.

### Experienced Behavior:
Nil pointer exception during illegal upgrade and broken upgrade functionality.

--- a/go.mod
+++ b/go.mod
@@ -12,7 +12,7 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/metal-stack/metal-go v0.32.0
 	github.com/metal-stack/metal-lib v0.17.0
-	github.com/metal-stack/updater v1.2.1
+	github.com/metal-stack/updater v1.2.2
 	github.com/metal-stack/v v1.0.3
 	github.com/olekukonko/tablewriter v0.0.6-0.20230925090304-df64c4bbad77
 	github.com/spf13/afero v1.11.0
@@ -157,7 +157,7 @@ require (
 	golang.org/x/net v0.26.0 // indirect
 	golang.org/x/oauth2 v0.21.0 // indirect
 	golang.org/x/sync v0.7.0 // indirect
-	golang.org/x/sys v0.21.0 // indirect
+	golang.org/x/sys v0.22.0 // indirect
 	golang.org/x/term v0.21.0 // indirect
 	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect

